src/reconcile/rec_write.c

```diff
@@ -2004,16 +2004,24 @@ __rec_split_write_reuse(
     if (mod->rec_result != WT_PM_REC_MULTIBLOCK || mod->mod_multi_entries < r->multi_next)
         return (false);
 
+    /*
+     * It is possible that the generated checksum values can be the same from two different images.
+     * Use the time aggregate also along with size and checksum in the comparison to identify
+     * whether the new image is the same as the previously written block to reuse.
+     */
     multi_match = &mod->mod_multi[r->multi_next - 1];
-    if (multi_match->size != multi->size || multi_match->checksum != multi->checksum) {
+    if (multi_match->size != multi->size || multi_match->checksum != multi->checksum ||
+      WT_TIME_AGGREGATE_IS_EMPTY(&multi->addr.ta) ||
+      WT_TIME_AGGREGATE_IS_EMPTY(&multi_match->addr.ta) ||
```

Comment:
Why did you include the empty as special cases here? Won't that mean that any reconciled image without time information will be replaced?

Reply:
Yes, the reconciled image without time information is not reused. I feel that Empty time aggregate may lead to the same situation of checksum being the same. Due to that reason, I added those special conditions.